packages/experimental-utils/src/index.ts

```diff
@@ -15,5 +15,6 @@ export {
   AST_NODE_TYPES,
   AST_TOKEN_TYPES,
   TSESTree,
+  TSESTreeToTSNode,
```

Comment:
I don't think we should expose it here unless someone needs it.
I can't think of a case where someone will need to use it directly.

Reply:
this is actually usefull in our rules:

```
type ConstructorTSType = TSESTreeToTSNode<TSESTree.MethodDefinition>;
```

in `prefer-readonly`

Reply:
```ts
  public enterConstructor(
    node: TSESTreeToTSNode<TSESTree.MethodDefinition>,
  ): void {
    // ......
  }
```

instead of 

```
  public enterConstructor(
    node:
      | ts.GetAccessorDeclaration
      | ts.SetAccessorDeclaration
      | ts.MethodDeclaration
      | ts.ConstructorDeclaration,
  ): void {
    // ......
  }
```

Reply:
than you prefer to keep it exposed?